pubsub/rabbitpubsub/rabbit.go

```diff
@@ -564,15 +589,34 @@ func (s *subscription) establishChannel(ctx context.Context) error {
 		if err != nil {
 			return err
 		}
+		// Apply subscription options to channel.
+		err = applyOptionsToChannel(s.opts, ch)
+		if err != nil {
+			return err
+		}
 		// Subscribe to messages from the queue.
 		s.delc, err = ch.Consume(s.queue, s.consumer)
 		return err
 	})
 	if err != nil {
 		return err
 	}
+
 	s.ch = ch
 	s.closec = ch.NotifyClose(make(chan *amqp.Error, 1)) // closec will get at most one element
+
+	return nil
+}
+
+func applyOptionsToChannel(opts *SubscriptionOptions, ch amqpChannel) error {
+	if opts == nil {
+		return nil
+	}
+
+	if err := ch.Qos(opts.PrefetchCount, 0, false); err != nil {
```

Comment:
Is it safe to set 0 as the default? What's the Rabbit default if we don't set it?

I think 0 is a valid value, which makes it a bad default :-|.

One approach is to add a separate boolean, eg.

int PrefetchCount
bool SetPrefetchCount

and only set it if SetPrefetchCount is true.

Another options is to use `int *PrefetchCount` and only set it if non-nil.

Reply:
done on 3cd7cf194059d19f9350be849630351dfdcda7d8